transport/src/main/java/io/netty/channel/embedded/EmbeddedChannel.java

```diff
@@ -723,7 +747,17 @@ protected void doClose() throws Exception {
 
     @Override
     protected void doBeginRead() throws Exception {
-        // NOOP
+        // read from tempInboundMessages and fire channelRead.
+        if (!tempInboundMessages().isEmpty()) {
```

Comment:
this means we will always create an `ArrayList` if if we never used `AUTO_READ` false before. We should do the same as we do with other queues here and use `isNotEmpty(tmpInboundMessages)`

Reply:
U are right and fixed.